#!/usr/bin/env python
PACKAGE = "lane_detect"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cluster_count",        int_t,   0, "KMean clustering",  3, 0, 10)

gen.add("offsetROI_x",  int_t,    0, "x offset on Hough lines",     30,   0, 320)
gen.add("offsetROI_y",  int_t,    0, "y position of the ROI box",   95,   0, 240)
gen.add("objectROI_h",  int_t,    0, "Height of the ROI box",       49,   0, 100)
gen.add("objectROI_offsetTop",         int_t,    0, "Move box to top x pixels", 45,  0, 240)
gen.add("diff_to_know_left_right",         int_t,    0, "% diff between left and right to classify", 40,  0, 100)
gen.add("depth_threshold_min",         int_t,    0, "Depth pixels close to the car", 10,  0, 255)
gen.add("depth_threshold_max",         int_t,    0, "Depth pixels far from the car", 155,  0, 255)

strategy_enum = gen.enum([
    gen.const("THRESHOLD_RAW_DEPTH",            int_t, 0, "Threshold on raw depth image"),
    gen.const("THRESHOLD_KMEAN",     int_t, 1, "Threshold on Kmean image"),
    gen.const("THRESHOLD_KMEAN_BGSUB",     int_t, 2, "Threshold on Kmean BGSUB image"),
    ], "Strategies")

gen.add("strategy", int_t, 0, "Strategy", 0, 0, 2, edit_method=strategy_enum)

exit(gen.generate(PACKAGE, "detect_object", "detectobject"))